buildkite/src/Pipeline/Mode.dhall

```diff
@@ -5,16 +5,17 @@
 
 let Prelude = ../External/Prelude.dhall
 
-let Mode = < PullRequest | Stable >
+let Mode = < PullRequest | Stable | PackageGeneration >
```

Comment:
Pipeline mode was originally designed for organizing pipelines based on frequency of runs or stage in which pipeline should . Here i think package generation pipeline is ran before hardfork, or frequently once per week. I would reflect that in name and not to mention purpose of pipeline mode

Reply:
Pipeline mode seems to be designed for selecting which pipeline something should be a part of, based on reading the code that filtering pipelines based on their mode. The PackageGeneration is not run on an interval, it is only run on-demand, so I've put it in it's own mode and updated the selection logic accordingly. Are you saying this is not the correct usage of the pipeline mode? I didn't quite understand your point about pipeline mode being for specifying the frequency or stage, can you expand more on that?

Reply:
Apologies i wasn't clear enough. In my previous comment i referred to frequency of runs but it is not relevant mode of pipeline. My understanding of pipeline mode is that it is basically a strategy according which we should pick up the jobs to be ran within pipeline. 

Initially 

| Pull Request | Stable |
| -----------------| ---------- |
| all defined **dirty** jobs | all defined jobs |


I've implemented Filtering mechanism some time a go which added more complexity and currently above modes can be translated to


| Pull Request | Stable |
| -----------------| ---------- |
| all defined **dirty** jobs tagged according to input filter | all defined jobs tagged according to input filter |

I think that PackageGeneration does not introduced another mode. It just uses Stable mode logic. The difference is only in used filter. 

It was my mistake not to rename properly modes after i added filtering stuff. It would be my pleasure to do it in separate PR on within this one. I think more adequate names would be Triaged or DirtyOnly  |  Filtered . Then such pipeline would be ran with env variables:

```
BUILDKITE_PIPELINE_MODE=Filtered
BUILDKITE_PIPELINE_FILTER=Release
```

Reply:
What I want at the end of the day is a way to filter out only the hardfork release jobs. I don't want the other jobs to run, and these hardfork jobs will never run in a PR.